elasticsearch/src/main/java/org/hibernate/search/elasticsearch/schema/impl/DefaultElasticsearchSchemaTranslator.java

```diff
@@ -281,6 +281,15 @@ private void addTypeOptions(PropertyMapping propertyMapping, BridgeDefinedField
 	private void addTypeOptions(PropertyMapping fieldMapping, FacetMetadata facetMetadata) {
 		ExtendedFieldType type;
 
+		if ( facetMetadata.isEncodingAuto() ) {
+			/*
+			 * If the user didn't ask for a specific encoding, just use the same datatype
+			 * as the source field.
+			 */
+			addTypeOptions( fieldMapping, facetMetadata.getSourceField() );
+			return;
+		}
+
```

Comment:
The issue is mainly that we don't have Lucene-specific metadata, so we're putting this lucene-specific encoding in the general metadata. Like we do for a lot of other things.

If we wanted to move the automatic encoding resolution around, there's no Lucene-specific, central place to store the resolved encoding, so the only thing we could do would be to do type resolution every time we need the actual encoding, which would be in `org.hibernate.search.engine.spi.DocumentBuilderIndexedEntity.addFacetDocValues(Document, DocumentFieldMetadata, FacetMetadata, Object)`. Which I find far more unappealing than adding that flag in the current metadata.
Note that it's not an issue of duplicating code (we only use the encoding in one place, it seems), but really more an issue of re-executing bootstrap code again and again at runtime.

Anyway, you're telling me "we need to break free of Lucene dependency", and that's definitely not the purpose of this PR, but more of 6.0. Once we have Lucene-independent APIs and (perhaps more importantly) engine, we'll add Lucene-specific metadata on top of engine metadata, as we will for Elasticsearch, and everything will be fine. Until then, well... we'll do as best we can.